packages/clay-card/src/Description.tsx

```diff
@@ -38,22 +43,24 @@ const ClayCardDescription: React.FunctionComponent<ICardDescriptionProps> = ({
 	truncate = true,
 	...otherProps
 }: ICardDescriptionProps) => {
-	const {interactive} = React.useContext(Context);
-
-	const interactiveTag = interactive ? 'span' : 'div';
-
-	const TagName = href ? 'a' : interactiveTag;
+	const OuterTag = CARD_TYPE_ELEMENTS[displayType] as ('span' | 'div' | 'h3');
```

Comment:
Just curious, why isn't TypeScript inferring the output here?

Reply:
It will infer the type of the value, `string`, but it wouldn't pick up the explicit string values.

Reply:
Oh yeah, this is true.

Reply:
@bryceosterhaus, @matuzalemsteles.

There are two ways to force it to infer a better type here.

Best way is to use [`as const`](https://github.com/Microsoft/TypeScript/pull/29510) like this:

```javascript
const CARD_TYPE_ELEMENTS = {
	subtitle: 'span',
	text: 'div',
	title: 'h3',
} as const;
```

This stops TypeScript from widening the type to `string` and you can drop the cast.

The other way that I can think of is to explicitly annotate the type, but that is very repetitive and I wouldn't suggest it:

```javascript
type T = {
	subtitle: 'span',
	text: 'div',
	title: 'h3',
};

const CARD_TYPE_ELEMENTS: T = {
	subtitle: 'span',
	text: 'div',
	title: 'h3',
};
```

[Playground link](https://www.typescriptlang.org/play/index.html#code/MYewdgzgLgBAwgQQEoBED6AVAmgBQKJp4AyeAsngHIYDKMAvDAN4BQAkBAK4BGUAllABsApgC4YAcggAHAIZhxAGjZQhADyhjxAE14A3Rcv7DNACwDMBgL4BuZsygBPKUPgyATlpRCIwN7yl84Ci80gIyDhhOLgziKuriMAA+EnyCQgnJktypwuK2zABmHGDAgWAwBWAAjAAUOqHhkc5icO6e3r7+ZcENEVEAlEzMMCMwoJCwAPIcKm4YMgDm9PDI6Nj4hCTkVNQA2vVSYX3OALowMhAwNZKy8kkSOvr34ubi-bajMG5CUBxu5dNZvMFrZLHZHM4YBhlix2NkjKIJNI5AZWHENA89KicoiXhYlDY7ONoCtUJhcARiGRKDRyfgUGJoQxYZweAjNMj5Eo0WoMdosdycaZ8cxCYViqVeOAKmAAEx1EKHRpRFptLw+PwBKVgHpK45CQYsT7EqYzIRzRbLRBk9aUrY06h0vAofaKo5NIQnD6jb6-f4wQHm4GgongEnWtYUzbUnZoOCTCjUDCMmFsVlCpG3bG8zSPbHsiSvAnnS4m-JFEplGXyg7ulWuDzqzpaoJu5XOQ3DUYmgNmi1LBgRp3R7a0+OJjCu3oer1dka+v4AvvB0XMIA)

Reply:
> Best way is to use as const like this...

This is a good approach. Thanks @wincent

Reply:
I sent [a PR](https://github.com/liferay/clay/pull/2241) with the suggested fix.